microprofile-open-api/src/test/java/org/jboss/eap/qe/microprofile/openapi/security/ListenerSecurityConfigurationTest.java

```diff
@@ -0,0 +1,238 @@
+        result.assertSuccess();
+
+        result = onlineManagementClient.execute(
+                "/subsystem=elytron/key-store=httpsGenKS:remove()");
+        result.assertSuccess();
+
+        new Administration(onlineManagementClient).reload();
+
+        Files.delete(keyStoreFile);
+    }
+
+    private static void disableHTTPListener() throws InterruptedException, TimeoutException, IOException, CliException {
+        ModelNodeResult result = onlineManagementClient.execute(
+                "/subsystem=undertow/server=default-server/http-listener=default:remove");
+        result.assertSuccess();
+
+        result = onlineManagementClient.execute(
+                "/subsystem=remoting/http-connector=http-remoting-connector:write-attribute(name=connector-ref, value=https)");
+        result.assertSuccess();
+
+        new Administration(onlineManagementClient).reload();
+    }
+
+    private static void enableHTTPListener() throws InterruptedException, TimeoutException, IOException, CliException {
+        ModelNodeResult result = onlineManagementClient.execute(
+                "/subsystem=undertow/server=default-server/http-listener=default:add(" +
+                        "socket-binding=http,redirect-socket=https,enable-http2=true)");
+        result.assertSuccess();
+
+        result = onlineManagementClient.execute(
+                "/subsystem=remoting/http-connector=http-remoting-connector:write-attribute(name=connector-ref, value=default)");
+        result.assertSuccess();
+
+        new Administration(onlineManagementClient).reload();
+    }
+
+    /**
+     * @tpTestDetails Test to verify that the {@code openapi} endpoint is available via HTTPS once the listener
+     *                has been properly configured
+     * @tpPassCrit The {@code openapi} endpoint can be reached via [https] schema at related standard port number
+     *             (8443), response status is HTTP 200 and response body is not empty
+     *
+     * @tpSince EAP 7.4.0.CD19
+     */
+    @Test
+    public void testOpenApiOnHTTPSListener() {
+        given().relaxedHTTPSValidation().get(String.format("%s?format=JSON", httpsOpenApiUrl))
+                .then()
+                .statusCode(200)
+                .contentType(equalToIgnoringCase(MediaType.APPLICATION_JSON))
+                .body(not(empty()));
+    }
+
+    /**
+     * @tpTestDetails Test to verify that activating MP OpenAPI when the HTTP (default) listener is disabled
+     *                causes the server to log a warning about MicroProfile OpenAPI specification requiring for the
+     *                endpoint to be accessible via [http]
+     * @tpPassCrit The server is logging expected {@code WFLYMPOAI0006} message because HTTP (default) listener
+     *             does not exist
+     *
+     * @tpSince EAP 7.4.0.CD19
+     */
+    @Test
+    public void testWarningIsLoggedBecauseOfMissingDefaultHTTPListener() {
```

Comment:
Though the method name clearly states what is does, could it be shortened somehow? For instance, `MissingHTTPListenerWarningLogged` or something similar? WDYT?

Reply:
Yes, sounds good to me. I'll use what you suggested leaving the `test` prefix in place. Fixing...